pkg/data/generic.go

```diff
@@ -119,34 +121,51 @@ func (s *GenericSync) RunContext(ctx context.Context) error {
 }
 
 // setup the store and queue for this GenericSync instance
-func (s *GenericSync) setup(ctx context.Context) (cache.Store, workqueue.DelayingInterface) {
+func (s *GenericSync) setup(ctx context.Context, ignoreNamespaces []string) (cache.SharedInformer, workqueue.DelayingInterface) {
```

Comment:
You'll want to pass the ignored namespaced to `NewFromInterface` rather than each method. That way you'll have it stored on the `GenericSync` object and won't have to change the interface of any current methods.